src/gl_limits.rs

```diff
@@ -29,26 +43,87 @@ impl Serialize for GLLimits {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
         where S: Serializer
     {
-        [self.max_vertex_attribs, self.max_tex_size, self.max_cube_map_tex_size]
+        [self.max_vertex_attribs, self.max_tex_size, self.max_cube_map_tex_size,
+         self.max_combined_texture_image_units, self.max_fragment_uniform_vectors,
+         self.max_renderbuffer_size, self.max_texture_image_units, self.max_varying_vectors,
+         self.max_vertex_texture_image_units, self.max_vertex_uniform_vectors]
             .serialize(serializer)
     }
 }
 
+macro_rules! gl_integer {
+    ($gl:ident, $pname:ident) => {
```

Comment:
what's a particular reason for this to be a macro as opposed to a function?

Reply:
The debugging output stringifies the name of the enum used.